bee-api/bee-rest-api/src/endpoints/routes/api/v2/message_children.rs

```diff
@@ -1,53 +1,34 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::net::IpAddr;
-
+use axum::{
+    extract::{Extension, Json, Path},
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
 use bee_message::MessageId;
-use bee_runtime::resource::ResourceHandle;
-use bee_tangle::Tangle;
-use warp::{filters::BoxedFilter, Filter, Rejection, Reply};
 
 use crate::{
-    endpoints::{
-        config::ROUTE_MESSAGE_CHILDREN, filters::with_tangle, path_params::message_id, permission::has_permission,
-        storage::StorageBackend,
-    },
+    endpoints::{error::ApiError, routes::api::v2::MAX_RESPONSE_RESULTS, storage::StorageBackend, ApiArgsFullNode},
     types::responses::MessageChildrenResponse,
 };
 
-fn path() -> impl Filter<Extract = (MessageId,), Error = warp::Rejection> + Clone {
-    super::path()
-        .and(warp::path("messages"))
-        .and(message_id())
-        .and(warp::path("children"))
-        .and(warp::path::end())
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/messages/:message_id/children", get(message_children::<B>))
 }
 
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    tangle: ResourceHandle<Tangle<B>>,
-) -> BoxedFilter<(impl Reply,)> {
-    self::path()
-        .and(warp::get())
-        .and(has_permission(ROUTE_MESSAGE_CHILDREN, public_routes, allowed_ips))
-        .and(with_tangle(tangle))
-        .and_then(message_children)
-        .boxed()
-}
-
-pub async fn message_children<B: StorageBackend>(
-    message_id: MessageId,
-    tangle: ResourceHandle<Tangle<B>>,
-) -> Result<impl Reply, Rejection> {
-    let mut children = Vec::from_iter(tangle.get_children(&message_id).await.unwrap_or_default());
+pub(crate) async fn message_children<B: StorageBackend>(
+    Path(message_id): Path<MessageId>,
+    Extension(args): Extension<ApiArgsFullNode<B>>,
+) -> Result<impl IntoResponse, ApiError> {
+    let mut children = Vec::from_iter(args.tangle.get_children(&message_id).await.unwrap_or_default());
     let count = children.len();
-    let max_results = 1000;
-    children.truncate(max_results);
-    Ok(warp::reply::json(&MessageChildrenResponse {
+    children.truncate(MAX_RESPONSE_RESULTS);
+
+    Ok(Json(MessageChildrenResponse {
         message_id: message_id.to_string(),
-        max_results,
+        max_results: MAX_RESPONSE_RESULTS,
         count,
         children_message_ids: children.iter().map(MessageId::to_string).collect(),
```

Comment:
You don't need to call `truncate`, just use `children.iter().take(MAX_RESPONSE_RESULTS).map(...).collect()`. You can then also get rid of the extra binding `count`.

Reply:
good point, done!